src/main/java/com/epam/code/mie/library/facades/BooksFacade.java

```diff
@@ -26,4 +26,9 @@ public List<BookDto> getAllBooks() {
     public Optional<BookDto> getBookByName(String name) {
         return booksService.getBookByName(name).map(booksMapper::toDto);
     }
+
+    @Transactional
+    public BookDto addBook(BookDto bookDto) {
+        return booksMapper.toDto(booksService.addBook(booksMapper.toEntity(bookDto)));
```

Comment:
Look for author by name, lastName and secondName here and set author to book entity only after that try to save Book entity